src/desktop/apps/consign/client/submission.tsx

```diff
@@ -72,28 +76,46 @@ function setupSubmissionFlow() {
                 exact
                 path="/consign/submission"
                 render={() => {
-                  if (sd.CURRENT_USER) {
+                  const params = queryString.parse(window.location.search)
+
+                  /**
+                   * If a logged out user arrives to consign, they first select
+                   * an artist and then are prompted to create an account.
+                   *
+                   * We push the artist id / name into query params so when the
+                   * page reloads (post user create) we can pre-populate the
+                   * artist step with previously selected artist info, and
+                   * skip directly to the "describe work" step.
+                   */
+                  if (params.artistId && params.artistName) {
+                    store.dispatch(
+                      chooseArtist({
+                        _id: params.artistId,
+                        name: params.artistName,
+                      })
+                    )
```

Comment:
This is the main change in the PR, which is storing state in query params and then plucking it off after the user has logged in / signed up. Everything else is basically just reordering array indexes so that sign in comes after select artist.